traffic_control/clients/python/to_access/__init__.py

```diff
@@ -0,0 +1,620 @@
+	                         "Note that this will make Content-Type headers \"wrong\", in general"))
+	parser.add_argument("PATH", help="The path to the resource being requested - omit '/api/1.x'")
+	parser.add_argument("DATA",
+	                    help=("An optional data string to pass with the request. If this is a "
+	                         "filename, the contents of the file will be sent instead."),
+	                    nargs='?')
+
+
+	args = parser.parse_args()
+
+	try:
+		to_host = args.to_host if args.to_host else os.environ["TO_URL"]
+	except KeyError as e:
+		raise KeyError("Traffic Ops hostname not set! Set the TO_URL environment variable or use "\
+		               "'--to_url'.") from e
+
+	useSSL = True
+	to_port = 443
+	if to_host.lower().startswith("http://"):
+		to_host = to_host[7:]
+		useSSL = False
+		to_port = 80
+	elif to_host.lower().startswith("https://"):
+		to_host = to_host[8:]
+
+	portpos = to_host.find(':')
+	if portpos > 0:
+		to_port = int(to_host[portpos+1:])
+		to_host = to_host[:portpos]
+
+	s = TOSession(to_host,
+	              host_port=to_port,
+	              ssl=useSSL,
+	              api_version=str(args.api_version),
+	              verify_cert=not args.insecure)
+	# s.create()
+
+	# The TOSession object's methods will handle '/' stripping for us, so it's only necessary with
+	# raw paths.
+	path = args.PATH if not args.raw_path else '/'.join((s.to_url.rstrip('/'), args.PATH.lstrip('/')))
+
+	data = args.DATA
+	if data and os.path.isfile(data):
+		with open(data) as f:
+			data = f.read()
+
+	if isinstance(data, str):
+		data = data.encode()
+
+	# if os.path.isfile(COOKIEFILE):
+	# 	with open(COOKIEFILE) as f:
+	# 		s._session.cookies.set("mojolicious", f.read())
+	# 	return s, path, data, args.full, args.raw_path
+
+	try:
+		to_user = args.to_user if args.to_user else os.environ["TO_USER"]
+	except KeyError as e:
+		raise KeyError("Traffic Ops user not set! Set the TO_USER environment variable or use "\
+		               "'--to_user'.") from e
+
+	try:
+		to_passwd = args.to_password if args.to_password else os.environ["TO_PASSWORD"]
+	except KeyError as e:
+		raise KeyError("Traffic Ops password not set! Set the TO_PASSWORD environment variable or "\
+		               "use '--to_password'") from e
+
+	try:
+		s.login(to_user, to_passwd)
+	except (OperationError, InvalidJSONError, LoginError) as e:
+		raise PermissionError from e
+
+	return (s,
+	       path,
+	       data,
+	       (
+	         args.request_headers or args.full,
+	         args.response_headers or args.full,
+	         args.request_payload or args.full
+	       ),
+	       args.raw_path,
+	       args.pretty)
+
+
+def get():
+	"""
+	Entry point for :program:`toget`
+
+	:returns: The program's exit code
+	"""
+	try:
+		s, path, data, full, raw, pretty = parse_arguments("toget")
+	except (PermissionError, KeyError) as e:
+		print(e, file=sys.stderr)
+		return 1
+
+	try:
+		if raw:
+			if data is not None:
```

Comment:
nit: generally it's more pythonic to omit `is not None` and just use `if data` here (same for the other line below this)

Reply:
That only works if you also want to skip the contained procedures _whenever_ `data` is a False-y value. In this case it could be an empty string, and while the meaning of that is unclear, the intention here is to reproduce exactly what is asked for on the command line.